src/osd/PG.cc

```diff
@@ -697,12 +697,10 @@ bool PG::MissingLoc::add_source_info(
 			 << ")" << dendl;
       continue;
     }
-    if (oinfo.last_complete < need) {
-      if (omissing.is_missing(soid)) {
-	ldout(pg->cct, 10) << "search_for_missing " << soid << " " << need
-			   << " also missing on osd." << fromosd << dendl;
-	continue;
-      }
+    if (omissing.is_missing(soid)) {
```

Comment:
Should we do this instead?

if (oinfo.last_complete < need || omissing.is_missing(soid)) {

Is it the case that if last_complete < need know we don't have the right version of the object, or if it is completely missing.  The is_missing() doesn't include the particular eversion_t.

Reply:
@dzafman I think we are already checking for that case earlier https://github.com/ceph/ceph/pull/26175/files#diff-dfb9ddca0a3ee32b266623e8fa489626R675

Reply:
Yes, I see.